src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -320,17 +306,15 @@ bool ObjectMonitor::enter_is_async_deflating() {
   return false;
 }
 
-void ObjectMonitor::enter_for_with_contention_mark(JavaThread* locking_thread, ObjectMonitorContentionMark& contention_mark) {
-  // Used by ObjectSynchronizer::enter_for to enter for another thread.
+bool ObjectMonitor::enterI_with_contention_mark(JavaThread* locking_thread, ObjectMonitorContentionMark& contention_mark) {
+  // Used by ObjectSynchronizer::enter_for() to enter for another thread.
```

Comment:
This renaming is confusing for me. The `enter_for` methods were made explicit because normally locking is always done by the current thread for the current thread - but deopt breaks that. And now it seems we have an `EnterI` that is really an `EnterI_for` ??

Reply:
Me too. So many functions that are sort of the same.

Reply:
As I wrote above, the confusing renaming has to be fixed.